extensions/resource-deployment/src/test/services/notebookService.test.ts

```diff
@@ -0,0 +1,294 @@
+import assert = require('assert');
+
+class Deferred<T> {
+	promise: Promise<T>;
+	resolve!: (value?: T | PromiseLike<T>) => void;
+	reject!: (reason?: any) => void;
+	constructor() {
+		this.promise = new Promise<T>((resolve, reject) => {
+			this.resolve = resolve;
+			this.reject = reject;
+		});
+	}
+}
+describe('NotebookService', function (): void {
+	const notebookInput = 'test-notebook.ipynb';
+	const sourceNotebookContent = '{ "cells": [] }';
+	const notebookFileName = 'mynotebook.ipynb';
+	const expectedTargetFileName = 'mynotebook';
+	const extensionPath = path.resolve(__dirname, '..', '..', '..');
+	const sourceNotebookRelativePath = `./notebooks/${notebookFileName}`;
+	const sourceNotebookAbsolutePath = path.resolve(extensionPath, sourceNotebookRelativePath);
+	const notebookWin32 = 'test-notebook-win32.ipynb';
+	const notebookDarwin = 'test-notebook-darwin.ipynb';
+	const notebookLinux = 'test-notebook-linux.ipynb';
+	const storagePath = __dirname;
+	let mockPlatformService: TypeMoq.IMock<IPlatformService>, notebookService: NotebookService;
+
+	beforeEach('NotebookService Setup', () => {
+		mockPlatformService = TypeMoq.Mock.ofType<IPlatformService>();//PlatformService, TypeMoq.MockBehavior.Loose
+		notebookService = new NotebookService(mockPlatformService.object, extensionPath);
+	});
+
+	afterEach('NotebookService cleanup', () => {
+		sinon.restore();
+		mockPlatformService.reset();
+	});
+
+	it('getNotebook with string parameter', async () => {
+		mockPlatformService.setup(x => x.fileExists(notebookInput)).returns(async () => true);
+		mockPlatformService.setup(x => x.readTextFile(notebookInput)).returns(async () => sourceNotebookContent);
+		let returnValue = await notebookService.getNotebook(notebookInput);
+		assert.deepEqual(returnValue, JSON.parse(sourceNotebookContent), 'returned notebook does not match expected value');
+	});
+
+	it('getNotebookPath with string parameter', () => {
+		mockPlatformService.setup((service) => service.platform()).returns(() => { return 'win32'; });
+		let returnValue = notebookService.getNotebookPath(notebookInput);
+		assert.strictEqual(returnValue, notebookInput, 'returned notebook name does not match expected value');
+		mockPlatformService.verify((service) => service.platform(), TypeMoq.Times.never());
+
+		mockPlatformService.reset();
+		mockPlatformService.setup((service) => service.platform()).returns(() => { return 'win32'; });
+		returnValue = notebookService.getNotebookPath('');
+		assert.strictEqual(returnValue, '', 'returned notebook name does not match expected value is not an empty string');
+		mockPlatformService.verify((service) => service.platform(), TypeMoq.Times.never());
+	});
+
+	it('getNotebook with NotebookInfo parameter', () => {
+		const notebookInput: NotebookPathInfo = {
+			darwin: notebookDarwin,
+			win32: notebookWin32,
+			linux: notebookLinux
+		};
+		mockPlatformService.setup((service) => service.platform()).returns(() => { return 'win32'; });
```

Comment:
Again - I'm really not a fan of forcing platforms like this. This one isn't really testing actual logic so it's ok I guess but I'd rather we just avoid all this kind of stuff in general unless there's a really good reason not to. Getting a couple extra lines of code coverage isn't really beneficial for the added test complexity and possible side effects of having a test pretend it's running on a certain platform that it isn't actually running on.

This could just be simplified to having all platforms returning the same notebook - that way you just need the one check.